docs/sphinx/specification/cudaq/types.rst

```diff
@@ -4,30 +4,91 @@ The CUDA Quantum language extension provides certain fundamental types that are
 to quantum-classical computing. These types are provided via defined classical library
 types in C++ in the :code:`cudaq` namespace. 
 
-:code:`cudaq::qudit<Levels>`
-----------------------------
-**[1]** The :code:`cudaq::qudit` models a :math:`D`-level unit of quantum information. The state of
-this system (for :math:`N` qudits) can be described by a :math:`D`\ :sup:`N`\-dimensional vector in
-Hilbert space with the absolute square of all elements summing to 1. 
+Quantum dit (qudit)
+===================
 
-**[2]** The :code:`cudaq::qudit` encapsulates a unique :code:`std::size_t` modeling the index of the
-qudit in the underlying quantum memory space (assuming an infinite register
-of available qudits). 
+The quantum dit, or qudit for short, is a model to represent a :math:`d`-level 
+(or :math:`d`-dimension) quantum system. Mathematically, we describe the state
+of a qudit using a :math:`d`-dimensional vector of unit form, living in the
+vector space :math:`\mathbb{C}^d` (a Hilbert space). Qudits with known values
+for :math:`d` have specific name. A qubit has dimension 2, a qutrit has
+dimension 3, and so on.
 
-**[4]** To adhere to the no-cloning theorem of quantum mechanics,
-the :code:`cudaq::qudit` is non-copyable and non-movable. Therefore, all :code:`cudaq::qudit` 
-instances must be passed by reference, and the no-cloning theorem is satisfied
-at compile-time. 
+In a CUDA Quantum implementation, objects of type qudit must adhere to the
+following constraints:
 
-**[5]** :code:`cudaq::qudit` instances can only be allocated within CUDA Quantum quantum
-kernel code and can never be allocated from classical host code.
+**[1]** Objects of type :code:`qudit` shall be parameterized with :math:`d`,
+        to indicate its dimension.
 
-**[6]** All instantiated :code:`cudaq::qudit` instances start in the :code:`|0>` computational basis 
-state and serve as the primary input argument for quantum intrinsic operations 
-modeling typical logical quantum gate operations.
+**[2]** CUDA Quantum kernels shall not have objects of type :code:`qudit` with
```

Comment:
I wonder if this should actually be a restriction in the kernels section of the document.